### PR TITLE
Add pre-send confirmation

### LINKED ISSUE #139: Pre-Send Confirmation
The custom implementation of the Publish/Send button has done away with pre-publish checklist, which makes it very easy to send a campaign, possibly without intending to. We should add some sort of confirmation UI, whether it's an "are you sure?" modal or a "ready to send" checkbox. We also could potentially do this only if the user has not sent themselves a test. 

> "will I ever not have a frisson of panic every time I hit the send button on the NL?" -- @kristastevens

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [X] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #139

### How to test the changes in this Pull Request:

1. Create a newsletter and fill settings so that it is valid ("Send" button is active) – set just list, without the group.
2. Click "Send" - observe a confirmation modal appears with the set list name and subscriber count
3. Cancel 
3. Set a group and click "Send" – observe that the modal now has information about the group and group's subscriber count
4. Send the campaign, observe that it is sent

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/editor/utils.js [hunk 1]
```diff
@@ -49,3 +54,44 @@ export const getEditPostPayload = ( {
...
+			label: title,
+			disabled: true,
+		} );
+		if ( interests && interests.interests && interests.interests.length ) {
+			interests.interests.forEach( interest => {
+				const isDisabled = parseInt( interest.subscriber_count ) === 0;
+				accumulator.push( {
+					label:
+						'- ' +
+						interest.name +
```
Comment (jeffersonrabb) [inline src/editor/utils.js:82]: While we're at it should the Interests/Groups `SelectControl` display subscriber count?

File: src/components/send-button/index.js [hunk 2]
```diff
@@ -82,22 +106,69 @@ export default compose( [
...
+				{ modalVisible && (
+					<Modal
+						className="newspack-newsletters__modal"
+						title={ __( 'Send your newsletter?', 'newspack-newsletters' ) }
+						onRequestClose={ () => setModalVisible( false ) }
+					>
+						<p>
+							{ __( "You're about to send a newsletter to:", 'newspack-newsletters' ) }
+							<br />
+							<b>{ listData.name }</b>
```
Comment (jeffersonrabb) [inline src/components/send-button/index.js:137]:
```suggestion
							<strong>{ listData.name }</strong>
```

File: src/components/send-button/index.js [hunk 3]
```diff
@@ -82,22 +106,69 @@ export default compose( [
...
+							<br />
+							{ listData.groupName && (
+								<Fragment>
+									{ __( 'Group:', 'newspack-newsletters' ) } <b>{ listData.groupName }</b>
+									<br />
+								</Fragment>
+							) }
+							<b>
```
Comment (jeffersonrabb) [inline src/components/send-button/index.js:145]:
```suggestion
							<strong>
```

File: src/components/send-button/index.js [hunk 4]
```diff
@@ -82,22 +106,69 @@ export default compose( [
...
+								{ sprintf(
+									_n(
+										'%d subscriber',
+										'%d subscribers',
+										listData.subscribers,
+										'newspack-newsletters'
+									),
+									listData.subscribers
+								) }
+							</b>
```
Comment (jeffersonrabb) [inline src/components/send-button/index.js:155]:
```suggestion
							</strong>
```

File: src/components/send-button/index.js (hunk 2 above)
Comment (adekbadek) [reply src/components/send-button/index.js]: done in acf67a54d4e759aaaa08eb65f0b5c94ca169475f

File: src/components/send-button/index.js (hunk 3 above)
Comment (adekbadek) [reply src/components/send-button/index.js]: done in acf67a54d4e759aaaa08eb65f0b5c94ca169475f

File: src/components/send-button/index.js (hunk 4 above)
Comment (adekbadek) [reply src/components/send-button/index.js]: done in acf67a54d4e759aaaa08eb65f0b5c94ca169475f

File: src/editor/utils.js (hunk 1 above)
Comment (adekbadek) [reply src/editor/utils.js]: Here you go: 5dc7c00a807466e1c195ede3898f6fc61ad1ea8e

**Pull request merged**